lib/ohai/plugins/cloudstack.rb

```diff
@@ -31,6 +33,11 @@
         cloudstack Mash.new
         Ohai::Log.debug("connecting to the 'cloudstack' metadata service")
         fetch_metadata.each { |k, v| cloudstack[k] = v }
+        # Note: the cloudstack public_ipv4 is the value of the NAT router (as per cloudstack documentation)
+        # and not necessarily the publicly available IP.  cloustack semi-supports floating
+        # ips in that the public ip for an instance can be an IP different from the NAT router
+        cloudstack['router_ipv4'] = cloudstack.delete('public_ipv4')
```

Comment:
From the description it sounds like this was a bug and we were violating the documented API for cloudstack.